Firestore/core/src/firebase/firestore/util/status.cc

```diff
@@ -69,11 +68,24 @@ Status& Status::WithPlatformError(std::unique_ptr<PlatformError> error) {
   return *this;
 }
 
+bool Status::IsMoved() const {
+  return IsStatePtrMoved(state_);
+}
+
+void Status::SetMoved() {
+  if (IsMoved()) {
+    return;
+  }
+
+  // Set pointer value to `0x1` as the pointer is no longer useful.
+  state_ = MakeStatePtrWithMovedIndicator();
```

Comment:
I'd refactor this slightly differently:
* create a function that returns the magic value:
```cpp
State* MovedFromIndicator() {
  return reinterpret_cast<State*>(0x01);
}
```
* use the `unique_ptr` constructor here directly:
```cpp
state_ = StatePtr{MovedFromIndicator()};
```
* when checking, do it like this:
```cpp
return state.get() == MovedFromIndicator();
```

Reply:
Err, #2 and #3 are still not implemented. Let me know if there's any trouble with these suggestions.